### ISSUE TITLE
Yolo training on google colab buffer over flow error

### ISSUE THREAD

**Issue by cristyioan2000** [pjreddie/darknet#1472]
I am trying to train a standard unmodified model using _yolov3-tiny.cfg_ . I am using Google Colab because I can't afford a good GPU to train my data on. I am training on 2 classes. and this error occurred..

### The command I pass to Colab:
`!./darknet detector train build/darknet/x64/data/obj.data build/darknet/x64/cfg/yolov3-tiny.cfg yolov3-tiny.weights -dont_show`

### The output I see
valid: Using default 'build/darknet/x64/data/train.txt'
yolov3-tiny
layer     filters    size              input                output
   0 conv     16  3 x 3 / 1   416 x 416 x   3   ->   416 x 416 x  16 0.150 BF
   1 max          2 x 2 / 2   416 x 416 x  16   ->   208 x 208 x  16 0.003 BF
   2 conv     32  3 x 3 / 1   208 x 208 x  16   ->   208 x 208 x  32 0.399 BF
   3 max          2 x 2 / 2   208 x 208 x  32   ->   104 x 104 x  32 0.001 BF
   4 conv     64  3 x 3 / 1   104 x 104 x  32   ->   104 x 104 x  64 0.399 BF
   5 max          2 x 2 / 2   104 x 104 x  64   ->    52 x  52 x  64 0.001 BF
   6 conv    128  3 x 3 / 1    52 x  52 x  64   ->    52 x  52 x 128 0.399 BF
   7 max          2 x 2 / 2    52 x  52 x 128   ->    26 x  26 x 128 0.000 BF
   8 conv    256  3 x 3 / 1    26 x  26 x 128   ->    26 x  26 x 256 0.399 BF
   9 max          2 x 2 / 2    26 x  26 x 256   ->    13 x  13 x 256 0.000 BF
  10 conv    512  3 x 3 / 1    13 x  13 x 256   ->    13 x  13 x 512 0.399 BF
  11 max          2 x 2 / 1    13 x  13 x 512   ->    13 x  13 x 512 0.000 BF
  12 conv   1024  3 x 3 / 1    13 x  13 x 512   ->    13 x  13 x1024 1.595 BF
  13 conv    256  1 x 1 / 1    13 x  13 x1024   ->    13 x  13 x 256 0.089 BF
  14 conv    512  3 x 3 / 1    13 x  13 x 256   ->    13 x  13 x 512 0.399 BF
  15 conv     21  1 x 1 / 1    13 x  13 x 512   ->    13 x  13 x  21 0.004 BF
  16 yolo
  17 route  13
  18 conv    128  1 x 1 / 1    13 x  13 x 256   ->    13 x  13 x 128 0.011 BF
  19 upsample            2x    13 x  13 x 128   ->    26 x  26 x 128
  20 route  19 8
  21 conv    256  3 x 3 / 1    26 x  26 x 384   ->    26 x  26 x 256 1.196 BF
  22 conv     21  1 x 1 / 1    26 x  26 x 256   ->    26 x  26 x  21 0.007 BF
  23 yolo
Total BFLOPS 5.449 
 Allocate additional workspace_size = 24.92 MB 
Loading weights from yolov3-tiny.weights...
 seen 64 
Done!
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Saving weights to backup//yolov3-tiny_final.weights
"
*** buffer overflow detected ***: ./darknet terminated

All I modified in the yolov3-tiny.cfg was _classes = 2, anchors, filters = 21_
I tried with a different cfg file, yolov3.cfg with no weights and got the same buffer overflow error

### Here is my cfg file:
```
[net]
# Testing
#batch=32
#subdivisions=2
#Training
batch=64
subdivisions=32
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=21
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=21
activation=linear



[yolo]
mask = 3,4,5
anchors =  59, 93,  88,158,  88,158,  88,158,  88,158,  88,158
classes=2
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=21
activation=linear

[yolo]
mask = 0,1,2
anchors =  59, 93,  88,158,  88,158,  88,158,  88,158,  88,158
classes=2
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
```

### obj.data 
Has the full path and it looks like this:
```
classes = 2
train = data/train.txt
names = data/obj.names
backup = backup/
```
Here is my colab file, with all the dependencies installed:
https://colab.research.google.com/drive/1kdwOl9AJ7A7Bn37YcrSVmwGH8U2EI_GC

**Comment by Wendong-yin** [comment]
I met the same issue, how do you solve this one?

**Comment by cristyioan2000** [comment]
I created the label files in windows and the VM from Colab is an Ubuntu so
i replaced all the ' \r ' with spaces or '' from the txt files and after
that i uploaded them on the VM and it worked.

În joi, 4 apr. 2019 la 00:15, eNeighbors <notifications@github.com> a scris:

**Comment by InputBlackBoxOutput** [comment]
I have the same problem but my annotation text files do not contain the '\r' character. Any suggestions on what I should do to debug this issue?

**Comment by cristyioan2000** [comment]
@InputBlackBoxOutput This error in my case appeared because I was creating/editing all the .txt files from Windows and I was using them on Linux. You need to run a dos2unix command on Linux or create all the text files you're using directly on Linux.
